app/src/main/java/com/android/unio/MainActivity.kt

```diff
@@ -46,7 +51,14 @@ fun UnioApp() {
     }
     navigation(startDestination = Screen.EXPLORE, route = Route.EXPLORE) {
       composable(Screen.EXPLORE) { ExploreScreen(navigationActions) }
-      composable(Screen.ASSOCIATION) { AssociationProfile(navigationActions) }
+      composable(Screen.ASSOCIATION_PROFILE + "/{uid}") { navBackStackEntry ->
+        val uid = navBackStackEntry.arguments?.getString("uid")
+        uid?.let { AssociationProfile(navigationAction = navigationActions, associationId = it) }
+            ?: run {
+              Log.e("AssociationProfile", "Association id is null")
+              Toast.makeText(context, "Association id is null", Toast.LENGTH_SHORT).show()
```

Comment:
This is very minor, but consider using UID instead of ID for consistence, and also written capitalized in the `Toast`.

Reply:
> This is very minor, but consider using UID instead of ID for consistence, and also written capitalized in the `Toast`.

Good catch I will change it.

> Also, this might be debatable but you could integrate the `+ "/{uid}"` directly in the `Screen.ASSOCIATION_PROFILE` constant.

Yes but I would argue that it doesn't fall into the category of a screen name, hence I wouldn't include it in the `Screen` enum. However, I think having an additional enum type for the screen arguments would be useful, don't you think?

Reply:
I see. Maybe we could have something like this:
```kotlin
enum class Screen(private val route: String) {
  HOME("home")
  ...
  ASSOCIATION("association/{uid}");

  fun withParams(vararg params: String): String {
    return params.fold(route) { acc, param -> acc.replaceFirst(Regex("\\{[^}]*}"), param) }
  }
}
```

Which could then be used like so:
```kotlin
navigationAction.navigateTo(ASSOCIATION.withParams("123"))
```

This would also work with nested routes with different UIDs due to the use of `varargs`.

Reply:
Wow, nice one, I wouldn't have thought of the use of `varargs`! Thanks for teaching me this. I see you used folding, like we did in Scala!